### DISCUSSION TITLE
File is always automatically saved as txt file

### DISCUSSION THREAD

**Discussion by bpezer** [linuxdeepin/developer-center#2951] [Features Request & Ideas | 特性请求 & 头脑风暴]
The editor allows the user to select different file types (scripts, source files, etc.), however even after changing the type the user needs to manually change the file extension from .txt to the appropriate one.

Choosing a file:
![filetype](https://user-images.githubusercontent.com/47884610/106912707-8e022480-6703-11eb-95ab-485f48ebc1aa.jpg)

The result of trying to save it:
![savefile](https://user-images.githubusercontent.com/47884610/106912757-9ce8d700-6703-11eb-91c0-d2380a54674f.jpg)

The file extension should be automatically changed by the editor when the user selects a certain file type.

**Comment by z966768** [comment]
The default file is txt, we judge according to the opening file.

**Comment by bpezer** [comment]
If that is the case a different solution might be to add other file extensions to the Format option in the Save As dialog:

![options](https://user-images.githubusercontent.com/47884610/107044162-3972ae00-67c4-11eb-803f-7060924f4100.jpg)

Making the user manually have to change file extensions for new files is a usability issue. Would this change be possible?

**Comment by guoshaoyu** [comment]
> The editor allows the user to select different file types (scripts, source files, etc.), however even after changing the type the user needs to manually change the file extension from .txt to the appropriate one.
> 
> Choosing a file:
> ![filetype](https://user-images.githubusercontent.com/47884610/106912707-8e022480-6703-11eb-95ab-485f48ebc1aa.jpg)
> 
> The result of trying to save it:
> ![savefile](https://user-images.githubusercontent.com/47884610/106912757-9ce8d700-6703-11eb-91c0-d2380a54674f.jpg)
> 
> The file extension should be automatically changed by the editor when the user selects a certain file type.

Thank you for your suggestion, we will consider this feature reasonably.
